sos/report/plugins/ssh.py

```diff
@@ -19,9 +19,28 @@ class Ssh(Plugin, IndependentPlugin):
     profiles = ('services', 'security', 'system', 'identity')
 
     def setup(self):
-        self.add_copy_spec([
+        sshcfgs = [
             "/etc/ssh/ssh_config",
             "/etc/ssh/sshd_config"
-        ])
+            ]
+
+        # Include main config files
+        self.add_copy_spec(sshcfgs)
+
+        # Read configs for any includes and copy those
+        try:
+            for sshcfg in sshcfgs:
+                with open(sshcfg, 'r') as cfgfile:
+                    for line in cfgfile.read().splitlines():
```

Comment:
This reads the entire content of the file into memory at once. In 99.9% of cases this will be fine for the expected files, but if this instead uses the form `for line in cfgfile`, we'll iterate line-by-line (what `.read().splitlines()` ends up doing anyways) in a safer and more efficient manner.

Reply:
Thank you for this information.  

I adjusted this to be:
```
   try:
            for sshcfg in sshcfgs:
                for line in open(sshcfg, 'r').read().splitlines():
```
Reran and plugin tested out OK grabbing the extra include files.  Please let me know if I misunderstood or if there are any other ideas on this.  Appreciate the assistance and review!

Reply:
The `.read().splitlines()` is what reads the entire file into memory at once, and is what I was suggesting to change. The `with` context manager is definitely beneficial as it ensures that the open file handle is always closed. This is what I was suggesting:

```python
        try:
            for sshcfg in sshcfgs:
                with open(sshcfg, 'r') as cfgfile:
                    for line in cfgfile:
```

If you use an open file in a `for` loop like this, python uses it as an iterator so we'll only read a single line at a time rather than reading everything into memory at once (`.read()`) and then splitting that into an iterable list (`.splitlines()`).

Reply:
Thanks for taking time to explain that further.  I didn't catch that it iterates this way but I see what you mean.  This makes sense and is more efficient.  
Made the adjustments and tested it out.  Thanks!